src/js/App/analytics/segment-analytics.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { createContext, useContext, useEffect, useRef } from 'react';
+import { AnalyticsBrowser } from '@segment/analytics-next';
+import { isProd } from '../../utils';
+import { useSelector } from 'react-redux';
+import { AnyObject } from '../../types';
+
+type SegmentEnvs = 'dev' | 'prod';
+type SegmentModules = 'openshift';
+
+const KEY_FALLBACK = {
+  prod: 'nm7VsnYsBVJ9MqjaVInft69pAkhCXq9Q',
+  dev: 'Aoak9IFNixtkZJRatfZG9cY1RHxbATW1',
+};
+
+function getAdobeVisitorId() {
+  const visitor = window?.s?.visitor;
+  if (visitor) {
+    return visitor.getMarketingCloudVisitorID();
+  }
+
+  return -1;
+}
+
+const getAPIKey = (env: SegmentEnvs = 'dev', module: SegmentModules) =>
+  ({
+    prod: {
+      openshift: 'z3Ic4EtzJtHrhXfpKgViJmf2QurSxXb9',
+    },
+    dev: {
+      openshift: 'A8iCO9n9Ax9ObvHBgz4hMC9htKB0AdKj',
+    },
+  }[env]?.[module] || KEY_FALLBACK[env]);
+
+const registerUrlObserver = () => {
+  /**
+   * We ignore hash changes
+   * Hashes only have frontend effect
+   */
+  let oldHref = document.location.href.replace(/#.*$/, '');
+
+  const bodyList = document.body;
+  const observer = new MutationObserver((mutations) => {
+    mutations.forEach(() => {
+      const newLocation = document.location.href.replace(/#.*$/, '');
+      if (oldHref !== newLocation) {
+        oldHref = newLocation;
+        window?.sendCustomEvent?.('pageBottom');
+        setTimeout(() => {
+          window.segment?.page(window?._segment?.pageOptions);
+        });
+      }
+    });
+  });
+  observer.observe(bodyList, {
+    childList: true,
+    subtree: true,
+  });
+  return observer.disconnect;
+};
+
+export const SegmentContext = createContext<{ ready: boolean; analytics?: AnalyticsBrowser }>({
+  ready: false,
+  analytics: undefined,
+});
+
+export type SegmentProviderProps = {
+  activeModule: string;
+};
+
+export const SegmentProvider: React.FC<SegmentProviderProps> = ({ activeModule, children }) => {
+  const analytics = useRef<AnalyticsBrowser>();
+  const user = useSelector(({ chrome: { user } }: AnyObject) => user);
+  useEffect(() => {
+    const disconnect = registerUrlObserver();
+    return () => disconnect();
+  }, []);
+
+  useEffect(() => {
+    if (activeModule && user) {
+      const newKey = getAPIKey(isProd() ? 'prod' : 'dev', activeModule as SegmentModules);
+      const identityOptions = { cloud_user_id: user.identity.internal.account_id, adobe_cloud_visitor_id: getAdobeVisitorId() };
+      const groupOptions = {
+        account_number: user.identity.account_number,
+        account_id: user.identity.internal.org_id,
+      };
+      if (!analytics.current) {
+        analytics.current = AnalyticsBrowser.load({ writeKey: newKey }, { initialPageview: false });
+        window.segment = analytics.current;
```

Comment:
We might want to set this as `window.analytics` or any apps trying to call `analytics.track` that are following the docs on Segment's website are probably going to be confused.

Reply:
The official API for applications is not going to be the window object. The `segmentAnalytics` object instance is exposed when using the `useChrome` hook. It is exposed here: https://github.com/RedHatInsights/insights-chrome/pull/1918/files#diff-90cf814c0a22d1dbb9311fa142c3e931efafebf0a19c9d58851cbdfcd9fd028fR107

I don't mind renaming the object key.